### PR TITLE
Image Labeling using Firebase ML in a Non-Expo React Native Application

### PR DESCRIPTION
**Pull request by zolomohan**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by zolomohan** [comment]
This fixes #1208

**Comment by hectorkambow** [comment]
@zolomohan 
Wanted to call attention to the high % in the 3rd party plagiarism checker. 

Not only does this tie back into original content - but also this may negatively affect the SEO of your article. 
[Image Labeling using Firebase ML in a Non-Expo React Native Application #1245.pdf](https://github.com/section-io/engineering-education/files/5839729/Image.Labeling.using.Firebase.ML.in.a.Non-Expo.React.Native.Application.1245.pdf)


We aim to hit 10% or lower.

**Comment by zolomohan** [comment]
@hectorkambow @WanjaMIKE 
About 10 matches from that list were from my previous react native article. I've already added citations to lines from documentations and Wikipedia.

Most of them are very minor stuff, since it's marking lines like "Add this line ..." or "Let's create a new project ..." as plagiarism.

File: articles/react-native-firebase-image-labelling/index.md [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+The Cloud Vision API allows developers to integrate vision detection features within applications like face and landmark detection, image labeling, optical character recognition (OCR), and tagging of explicit content. [Cloud Vision Docs](https://cloud.google.com/vision/docs).
+
+### Firebase
+Firebase is a platform developed by Google for creating mobile and web applications. It was originally an independent company founded in 2011. In 2014, Google acquired the platform and it is now their flagship offering for app development. [Wikipedia](https://en.wikipedia.org/wiki/Firebase)
+
+### Prerequisites
+
+To proceed with this tutorial:
+
+- You will need a fundamenal knowledge of React & React Native.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-image-labelling/index.md:28]: fundamental

File: articles/react-native-firebase-image-labelling/index.md [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+author: mohan-raj
+date: 2021-01-07T00:00:00-15:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/react-native-image-labeling-with-firebase/hero.jpg
+    alt: React Native Image Labeling With Firebase Hero Image
+---
+Image labeling gives you insight into the content of images. In this tutorial, we will be building a Non-Expo React Native application to label the image provided using Firebase's machine learning kit.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-image-labelling/index.md:17]: This section should not contain this `can't` but rather `cannot`. affected word is `Firebase's` Ref peer review documentation

File: articles/react-native-firebase-image-labelling/index.md [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+
+The Cloud Vision API allows developers to integrate vision detection features within applications like face and landmark detection, image labeling, optical character recognition (OCR), and tagging of explicit content. [Cloud Vision Docs](https://cloud.google.com/vision/docs).
+
+### Firebase
+Firebase is a platform developed by Google for creating mobile and web applications. It was originally an independent company founded in 2011. In 2014, Google acquired the platform and it is now their flagship offering for app development. [Wikipedia](https://en.wikipedia.org/wiki/Firebase)
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-image-labelling/index.md:22]: Provide some explanation of Firebase ML.

File: articles/react-native-firebase-image-labelling/index.md [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+```JSX
+image: {
+  height: 300,
+  width: 300,
+  marginTop: 20,
+  borderRadius: 10,
+},
+```
+
+If the user did not cancel the operation, let's set the image state with the URI of the selected image in the onImageSelect function.
```
Comment (WanjaMIKE) [inline articles/react-native-firebase-image-labelling/index.md:316]: `If the user did not cancel the operation, let's set the image state with the URI of the selected image in the onImageSelect function.`

This sounds better

`If the user did not cancel the operation, we should set the image state with the URI of the selected image in the onImageSelect function.`

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
@zolomohan Nice Article. Quite informative. Just a few changes

**Comment by zolomohan** [comment]
@WanjaMIKE Hey @WanjaMIKE. Thanks for the review. I've made those minor changes.

**Comment by WanjaMIKE** [comment]
Okay. Also, you do reuse sections from previous articles. Right?

**Comment by zolomohan** [comment]
> Okay. Also, you do reuse sections from previous articles. Right?

Yep. The common steps 😅

**Comment by WanjaMIKE** [comment]
@zolomohan Just finished the tech review. The code works as expected. However, I noted that your images links my not be correct.
your images have this - `/engineering-education/react-native-image-labeling-with-firebase/`
while the branch has this name -` react-native-firebase-image-labelling` 
The links, therefore, appear to be wrong

**Comment by zolomohan** [comment]
The image URLs are supposed to be the URLs in the front matter, not the branch name.

**Comment by zolomohan** [comment]
@WanjaMIKE I've updated the Image URLs as per the front-matter docs.

**Review by WanjaMIKE — APPROVED** [review]

**Pull request merged**